modules/nf-core/parabricks/fq2bammeth/tests/main.nf.test

```diff
@@ -49,11 +50,73 @@ nextflow_process {
         then {
             assertAll(
                 { assert process.success },
-                { assert snapshot(process.out).match() }
+                { assert snapshot(
+                    bam(process.out.bam[0][1]).getReadsMD5(),
+                    process.out.bai,
+                    process.out.versions
+                ).match() }
             )
         }
+    }
+
+    test("SRR389222 - fastq - se - stub") {
+        options '-stub'
 
+        when {
+            process {
+                """
+                input[0] = Channel.of([
+				    [ id:'test', single_end:true ], // meta map
+				    [
+				        file('https://github.com/nf-core/test-datasets/raw/methylseq/testdata/SRR389222_sub1.fastq.gz', checkIfExists: true)
+				    ]
+				])
+                input[1] = Channel.of([
+                    [ id:'test' ], // meta map
+                    file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/genome.fa', checkIfExists: true)
+                ])
+				input[2] = BWAMETH_INDEX.out.index
+                input[3] = []
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
+            )
+        }
     }
 
+    test("E. coli - fastq - pe - stub") {
```

Comment:
This is still not working without `stub`. It throws signal 11 error. I am investigating.

Reply:
Error log,

```bash
Caused by:
    Process `PARABRICKS_FQ2BAMMETH (test)` terminated with an error exit status (255)
  
  
  Command executed:
  
    ln -sf $(readlink genome.fa) BwamethIndex/genome.fa
    
    pbrun \
        fq2bam_meth \
        --ref BwamethIndex/genome.fa \
        --in-fq Ecoli_10K_methylated_R1.fastq.gz Ecoli_10K_methylated_R2.fastq.gz \
        --out-bam test.bam \
         \
         \
         \
        --low-memory \
        --verbose \
        --x3
    
    cat <<-END_VERSIONS > versions.yml
    "PARABRICKS_FQ2BAMMETH":
        pbrun: $(echo $(pbrun version 2>&1) | sed 's/^Please.* //' )
    END_VERSIONS
  
  Command exit status:
    255
  
  Command output:
    Please visit https://docs.nvidia.com/clara/#parabricks for detailed documentation
    
    
    /usr/local/parabricks/run_pb.py fq2bam_meth --ref BwamethIndex/genome.fa --in-fq Ecoli_10K_methylated_R1.fastq.gz Ecoli_10K_methylated_R2.fastq.gz --out-bam test.bam --low-memory --verbose --x3 --tmp-dir EP7NBVV4
    
    [Parabricks Options Mesg]: Checking argument compatibility
    [Parabricks Options Mesg]: Set --bwa-options="-K #" to produce compatible pair-ended results with previous versions of
    fq2bam or BWA MEM.
    [Parabricks Options Mesg]: Automatically generating ID prefix
    [Parabricks Options Mesg]: Read group created for /home/ubuntu/nf-modules/.nf-test/tests/e54c69381087ebf278ad7a5755bf1ac
    1/work/stage-6589203e-a602-4940-be4e-03532cb9b4b7/98/df175278a211472ef880f0d2d7b178/Ecoli_10K_methylated_R1.fastq.gz and
    /home/ubuntu/nf-modules/.nf-test/tests/e54c69381087ebf278ad7a5755bf1ac1/work/stage-6589203e-a602-4940-be4e-
    03532cb9b4b7/ec/8f7827667945e06a0a9e9bddbe9cf0/Ecoli_10K_methylated_R2.fastq.gz
    [Parabricks Options Mesg]:
    @RG\tID:1_gi|170079663|ref|NC_010473.1.1\tLB:lib1\tPL:bar\tSM:sample\tPU:1_gi|170079663|ref|NC_010473.1.1
    [Parabricks Options Mesg]: Using --low-memory reduces the number of reads sent to GPU per batch in fq2bam_meth.
    [Parabricks Options Mesg]: Using --low-memory sets the number of streams in bwa mem to 1.
    /usr/local/parabricks/binaries/bin/pbbwa /home/ubuntu/nf-modules/.nf-test/tests/e54c69381087ebf278ad7a5755bf1ac1/work/75/c405addfb2a5e769378b9ad83f3c08/BwamethIndex/genome.fa.bwameth.c2t --mode pair-ended-gpu /home/ubuntu/nf-modules/.nf-test/tests/e54c69381087ebf278ad7a5755bf1ac1/work/stage-6589203e-a602-4940-be4e-03532cb9b4b7/98/df175278a211472ef880f0d2d7b178/Ecoli_10K_methylated_R1.fastq.gz /home/ubuntu/nf-modules/.nf-test/tests/e54c69381087ebf278ad7a5755bf1ac1/work/stage-6589203e-a602-4940-be4e-03532cb9b4b7/ec/8f7827667945e06a0a9e9bddbe9cf0/Ecoli_10K_methylated_R2.fastq.gz -R @RG\tID:1_gi|170079663|ref|NC_010473.1.1\tLB:lib1\tPL:bar\tSM:sample\tPU:1_gi|170079663|ref|NC_010473.1.1 --nGPUs 1 --nstreams 1 --cpu-thread-pool 16 -T 40 -B 2 -L 10 -U 100 -C -M --meth-mode --reduce-batch-size -F 0 --min-read-size 1 --max-read-size 480 --markdups --write-bin --verbose
    For technical support visit https://docs.nvidia.com/clara/index.html#parabricks
    Exiting...
    
    Could not run fq2bam_meth
    Exiting pbrun ...
  
  Command error:
    Please visit https://docs.nvidia.com/clara/#parabricks for detailed documentation
    
    
    /usr/local/parabricks/run_pb.py fq2bam_meth --ref BwamethIndex/genome.fa --in-fq Ecoli_10K_methylated_R1.fastq.gz Ecoli_10K_methylated_R2.fastq.gz --out-bam test.bam --low-memory --verbose --x3 --tmp-dir EP7NBVV4
    
    [Parabricks Options Mesg]: Checking argument compatibility
    [Parabricks Options Mesg]: Set --bwa-options="-K #" to produce compatible pair-ended results with previous versions of
    fq2bam or BWA MEM.
    [Parabricks Options Mesg]: Automatically generating ID prefix
    [Parabricks Options Mesg]: Read group created for /home/ubuntu/nf-modules/.nf-test/tests/e54c69381087ebf278ad7a5755bf1ac
    1/work/stage-6589203e-a602-4940-be4e-03532cb9b4b7/98/df175278a211472ef880f0d2d7b178/Ecoli_10K_methylated_R1.fastq.gz and
    /home/ubuntu/nf-modules/.nf-test/tests/e54c69381087ebf278ad7a5755bf1ac1/work/stage-6589203e-a602-4940-be4e-
    03532cb9b4b7/ec/8f7827667945e06a0a9e9bddbe9cf0/Ecoli_10K_methylated_R2.fastq.gz
    [Parabricks Options Mesg]:
    @RG\tID:1_gi|170079663|ref|NC_010473.1.1\tLB:lib1\tPL:bar\tSM:sample\tPU:1_gi|170079663|ref|NC_010473.1.1
    [Parabricks Options Mesg]: Using --low-memory reduces the number of reads sent to GPU per batch in fq2bam_meth.
    [Parabricks Options Mesg]: Using --low-memory sets the number of streams in bwa mem to 1.
    /usr/local/parabricks/binaries/bin/pbbwa /home/ubuntu/nf-modules/.nf-test/tests/e54c69381087ebf278ad7a5755bf1ac1/work/75/c405addfb2a5e769378b9ad83f3c08/BwamethIndex/genome.fa.bwameth.c2t --mode pair-ended-gpu /home/ubuntu/nf-modules/.nf-test/tests/e54c69381087ebf278ad7a5755bf1ac1/work/stage-6589203e-a602-4940-be4e-03532cb9b4b7/98/df175278a211472ef880f0d2d7b178/Ecoli_10K_methylated_R1.fastq.gz /home/ubuntu/nf-modules/.nf-test/tests/e54c69381087ebf278ad7a5755bf1ac1/work/stage-6589203e-a602-4940-be4e-03532cb9b4b7/ec/8f7827667945e06a0a9e9bddbe9cf0/Ecoli_10K_methylated_R2.fastq.gz -R @RG\tID:1_gi|170079663|ref|NC_010473.1.1\tLB:lib1\tPL:bar\tSM:sample\tPU:1_gi|170079663|ref|NC_010473.1.1 --nGPUs 1 --nstreams 1 --cpu-thread-pool 16 -T 40 -B 2 -L 10 -U 100 -C -M --meth-mode --reduce-batch-size -F 0 --min-read-size 1 --max-read-size 480 --markdups --write-bin --verbose
    [PB Info 2024-Oct-26 23:03:13] ------------------------------------------------------------------------------
    [PB Info 2024-Oct-26 23:03:13] ||                 Parabricks accelerated Genomics Pipeline                 ||
    [PB Info 2024-Oct-26 23:03:13] ||                              Version 4.3.2-1                             ||
    [PB Info 2024-Oct-26 23:03:13] ||                      GPU-PBBWA mem, Sorting Phase-I                      ||
    [PB Info 2024-Oct-26 23:03:13] ------------------------------------------------------------------------------
    [PB Debug 2024-Oct-26 23:03:13][src/main.cpp:235] Read in 1 fq pairs
    [PB Info 2024-Oct-26 23:03:13] Mode = pair-ended-gpu
    [PB Info 2024-Oct-26 23:03:13] Running with 1 GPU(s), using 1 stream(s) per device with 16 worker threads per GPU
    [PB Error 2024-Oct-26 23:03:13][-unknown-:0] Received signal: 11
    For technical support visit https://docs.nvidia.com/clara/index.html#parabricks, exiting.
    For technical support visit https://docs.nvidia.com/clara/index.html#parabricks
    Exiting...
    
    Could not run fq2bam_meth
    Exiting pbrun ...
```